README.md

```diff
@@ -221,6 +221,7 @@ In a module annotated with `#[ink::contract]` these attributes are available:
 | `#[ink(constructor)]` | Applicable to method. | Flags a method for the ink! storage struct as constructor making it available to the API for instantiating the contract. |
 | `#[ink(payable)]` | Applicable to ink! messages. | Allows receiving value as part of the call of the ink! message. ink! constructors are implicitly payable. |
 | `#[ink(selector = "..")]` | Applicable to ink! messages and ink! constructors. | Specifies a concrete dispatch selector for the flagged entity. This allows a contract author to precisely control the selectors of their APIs making it possible to rename their API without breakage. |
+| `#[ink(label = "..")]` | Applicable to ink! messages, ink! constructors and ink! implementation blocks. | Specifies a concrete label of method/trait inside of metadata. This allows a contract author to precisely control the labeling of methods inside of ABI making it possible to rename their methods without breakage. |
```

Comment:
```suggestion
| `#[ink(label = "..")]` | Applicable to ink! messages, ink! constructors and ink! implementation blocks. | Specifies a concrete label of a method/trait inside the metadata. This allows a contract author to precisely control the labeling of methods inside of ABI making it possible to rename their methods without breakage. |
```

Reply:
> | `#[ink(label = "..")]` | Applicable to ink! messages, ink! constructors and ink! implementation blocks. 

Why do we have the need to label ink! implementation blocks? What is the effect of it?

Reply:
Marking the implementation block allows you to specify the prefix before the message or constructor. 
The user can use it to remove the prefix of trait, or change the prefix of trait, or add a prefix for the impl section without the trait.

In this [PR](https://github.com/paritytech/ink/pull/859) I described the case when we are using "change the prefix of trait" behavior.

Reply:
I am not sure I like this feature for trait implementation blocks since I see a potential for a lot of confusion. What is an actual concrete use case there?

Reply:
The actual use case is:
We have a trait with the name `Erc20External`. During generation of labels for methods ink! adds `Erc20External::` prefix. With this prefix it generates `Erc20External::transfer` label. The idea is to label the impl section with the `Erc20` label to generate the `Erc20::transfer` for the method.

We can remove the `label` attribute for the impl section if the `label` attribute also overrides the prefix(in the current implementation it only overrides the label of method and trait prefix affects it).

Reply:
> I am not sure I like this feature for trait implementation blocks since I see a potential for a lot of confusion. What is an actual concrete use case there?

@Robbepop Can you elaborate on where you see potential for confusion here? If you haven't seen yet, the [issue for this PR](https://github.com/paritytech/ink/issues/900) states some more of the motivation:

>For traits, ink! generates a field in the form of `"name": ["BaseErc20", "new"]`. The `BaseErc20` hereby is a trait implemented by the contract. This exposes internal implementation details of the contract without the developer having any influence over it.